src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -946,6 +1019,24 @@ public static URL checkEndPoint(String url) throws FormValidation {
         }
     }
 
+    @CheckForNull
+    private static BasicSSHUserPrivateKey getSshCredential(String id){
+
+        BasicSSHUserPrivateKey credential = CredentialsMatchers.firstOrNull(
+                CredentialsProvider.lookupCredentials(
+                        BasicSSHUserPrivateKey.class, // (1)
```

Comment:
[As noted in Jira](https://issues.jenkins.io/browse/JENKINS-56927?focusedId=398071&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-398071) this is incorrect: should be using the interface type, `SSHUserPrivateKey`.

Reply:
I see #602 changing other things. I think you meant #520.

Reply:
Oh yes you're right!